src/Tools/ExternalAccess/Razor/IRazorCapabilitiesProvider.cs

```diff
@@ -2,12 +2,22 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
 {
+    [Obsolete("Use IRazorTestCapabilitiesProvider")]
     internal interface IRazorCapabilitiesProvider
     {
         ServerCapabilities GetCapabilities(ClientCapabilities clientCapabilities);
     }
+
+    /// <summary>
+    /// A capabilities provider that should only be used from Razor tests
```

Comment:
I wonder if there's anything we can use to ban the usage of any LSP protocol types in this project.  Maybe ban the entire LSP protocol namespace?
https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md

Reply:
I thought about this too, but the issue is that we don't want to ban the usage, just the exposure. In fact we _require_ the usage, because thats what the Roslyn LSP server we're interacting with needs.